compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4702,5 +4716,37 @@ private boolean isUnaryOperator(SyntaxKind kind) {
                 return false;
         }
     }
+
+    /**
+     * Parse is expression.
+     * <code>
+     * is-expr := expression is type-descriptor
+     * </code>
+     *
+     * @param lhsExpr Preceding expression of the is expression
+     * @return Is expression node
+     */
+    private STNode parseIsExpression(STNode lhsExpr) {
+        STNode isKeyword = parseIsKeyword();
+        startContext(ParserRuleContext.IS_EXPRESSION);
```

Comment:
Does these two lines needs to be switched?

Reply:
Do we need to start a context at all?

Reply:
Without a starting context,  `getNextRuleForSimpleTypeDesc()` looks for a 'VARIABLE_NAME'. But for `is_expression` we need to have `expression_rhs` as the next rule right?

Reply:
ack.